api/src/friends/friends.handlers.ts

```diff
@@ -217,39 +164,29 @@ export const requestAddFriend = async (
       return;
     }
 
-    const [existingOppositeRequest] = await tx
-      .selectDistinctOn([
-        studentFriendRequests.netId,
-        studentFriendRequests.friendNetId,
-      ])
-      .from(studentFriendRequests)
-      .where(
-        and(
-          eq(studentFriendRequests.netId, friendNetId),
-          eq(studentFriendRequests.friendNetId, netId),
-        ),
-      );
-
-    if (existingOppositeRequest) {
-      res.status(400).json({ error: 'ALREADY_RECEIVED_REQUEST' });
-      return;
-    }
-
-    const [existingSameRequest] = await tx
+    const [existingRequest] = await tx
       .selectDistinctOn([
         studentFriendRequests.netId,
         studentFriendRequests.friendNetId,
       ])
       .from(studentFriendRequests)
       .where(
-        and(
-          eq(studentFriendRequests.netId, netId),
-          eq(studentFriendRequests.friendNetId, friendNetId),
+        or(
+          and(
+            eq(studentFriendRequests.netId, netId),
+            eq(studentFriendRequests.friendNetId, friendNetId),
+          ),
+          and(
+            eq(studentFriendRequests.netId, friendNetId),
+            eq(studentFriendRequests.friendNetId, netId),
+          ),
```

Comment:
Is there ever a case where both exist? If this is the case, then the array destructuring won't work as intended